app/lib/pre_assembly/batch.rb

```diff
@@ -54,11 +54,12 @@ def digital_objects
 
       containers_via_manifest.map.with_index do |container, i|
         stageable_items = discover_items_via_crawl(container)
+        common_path = Assembly::ObjectFile.common_path(stageable_items) # find common paths to all files provided
         row = object_manifest_rows[i]
         yield DigitalObject.new(self,
                                 container: container,
                                 stageable_items: stageable_items,
-                                object_files: ObjectFileFinder.run(stageable_items: stageable_items, druid: row[:druid]),
+                                object_files: stageable_items.map { |item| PreAssembly::ObjectFile.new(item, { relative_path: item.gsub(common_path, '') }) },
                                 label: row.fetch('label', ''),
```

Comment:
The removed `ObjectFileFinder` class simply iterated through each stageable item and then globbed it for files (and excluded directories).  However, the original glob that populates stageable items (shown below) already finds all of the files recursively, so we can simply exclude directories at that point and not do another search.  We can also correctly and easily compute the relative path of each file by using our existing method for finding the path common to all stageable files and removing it).